cuda/base/cusolver_bindings.hpp

```diff
@@ -0,0 +1,139 @@
+// SPDX-FileCopyrightText: 2017 - 2025 The Ginkgo authors
+//
+// SPDX-License-Identifier: BSD-3-Clause
+
+#ifndef GKO_CUDA_BASE_CUSOLVER_BINDINGS_HPP_
+#define GKO_CUDA_BASE_CUSOLVER_BINDINGS_HPP_
+
+
+#include <cuda.h>
+#include <cusolverDn.h>
+
+#include <ginkgo/core/base/exception_helpers.hpp>
+
+#include "common/cuda_hip/base/types.hpp"
+
+
+namespace gko {
+namespace kernels {
+namespace cuda {
+/**
+ * @brief The CUSOLVER namespace.
+ *
+ * @ingroup cusolver
+ */
+namespace cusolver {
+/**
+ * @brief The detail namespace.
+ *
+ * @ingroup detail
+ */
+namespace detail {
+
+
+template <typename... Args>
+inline int64 not_implemented(Args...)
+{
+    return static_cast<int64>(CUSOLVER_STATUS_MATRIX_TYPE_NOT_SUPPORTED);
```

Comment:
it should throw the Ginkgo error not the cusolver one. the function return will not be caught.

Reply:
This is what the cuSPARSE and cuBLAS bindings do, too, though, right?  I was trying to follow those.

Reply:
you are right. I read the macro binding wrongly.